### ISSUE TITLE
Not able to connect to bigchaindb with docker

### ISSUE THREAD

**Issue by jeanjluo** [bigchaindb/bigchaindb#1407]
ubuntu 14.04
docker 17.03.1-ce

I am following the online doc for running BigchainDB with docker: https://docs.bigchaindb.com/projects/server/en/latest/appendices/run-with-docker.html and got a connection error: 

`requests.exceptions.ConnectionError: HTTPConnectionPool(host='172.17.0.1', port=9984): Max retries exceeded with url: /api/v1/transactions/ (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7f8057d35390>: Failed to establish a new connection: [Errno 111] Connection refused',))`

The bigchaindb keeps restarting:

> CONTAINER ID        IMAGE                   COMMAND                  CREATED              STATUS                          PORTS                         NAMES
> ae36f28bae33        bigchaindb/bigchaindb   "bigchaindb --dev-..."   About a minute ago   Restarting (1) 35 seconds ago                                 bigchaindb
> 1aad93dcc472        mongo:3.4.1             "/entrypoint.sh --..."   20 hours ago         Up 20 hours                     172.17.0.1:27017->27017/tcp   mongodb

Also there is no "blockchain" db created in mongodb:

> bigchain-rs:PRIMARY> show databases
> admin 0.000GB
> local 0.000GB

**Comment by krish7919** [comment]
@jeanjluo : Some package with locale dependencies have broken recently (11 days ago), which were causing automated docker builds to fail.

I have this temporary workaround for now that you can use to build locally:
```
git diff
diff --git a/Dockerfile b/Dockerfile
index 021f677..9df3037 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -7,8 +7,9 @@ RUN mkdir -p /usr/src/app
 COPY . /usr/src/app/
 WORKDIR /usr/src/app
 
-RUN locale-gen en_US.UTF-8 && \
-    apt-get -q update && \
+RUN apt-get -q update && \
+    apt-get install -y locales && \
+    locale-gen en_US.UTF-8 && \
     apt-get install -qy --no-install-recommends \
         python3 \
         python3-pip \
```

**Comment by jeanjluo** [comment]
The bigchaindb container started normal, and I am able to see the "bigchain" db in mongodb now:

> bigchain-rs:PRIMARY> show databases
> admin     0.000GB
> bigchain  0.000GB
> local     0.000GB

However, I am still getting connection error calling the container from the same host:

> requests.exceptions.ConnectionError: HTTPConnectionPool(host='172.17.0.1', port=9984): Max retries exceeded with url: /api/v1/transactions/ (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7f1a2672d7f0>: Failed to establish a new connection: [Errno 111] Connection refused',))

```
from bigchaindb_driver import BigchainDB
bdb = BigchainDB('http://172.17.0.1:9984')

bicycle = {
  'data': {
    'bicycle': {
      'serial_number': 'abcd1234',
      'manufacturer': 'bkfab',
    },
   },
}

metadata = {'planet': 'earth'}

from bigchaindb_driver.crypto import generate_keypair
alice, bob = generate_keypair(), generate_keypair()

prepared_creation_tx = bdb.transactions.prepare(
  operation='CREATE',
  signers=alice.public_key,
  asset=bicycle,
  metadata=metadata,
)

fulfilled_creation_tx = bdb.transactions.fulfill(
  prepared_creation_tx, private_keys=alice.private_key)

sent_creation_tx = bdb.transactions.send(fulfilled_creation_tx)
```

ifconfig:

> docker0   Link encap:Ethernet  HWaddr 02:42:a6:7a:7c:a0  
>           inet addr:172.17.0.1  Bcast:0.0.0.0  Mask:255.255.0.0

**Comment by jeanjluo** [comment]
I figured out i should be using the IP of the container in bridge. It is working now. Thanks !

**Comment by krish7919** [comment]
Hi @jeanjluo,
We have released v0.10.1 which includes a fix for the docker build.
